### DISCUSSION TITLE
Json.schema - method apply in trait Schema is deprecated

### DISCUSSION THREAD

**Discussion by michalbogacz** [andyglow/scala-jsonschema#217] [Q&A] [answered]
Hello,

I'm using libs
```
      "com.github.andyglow" %% "scala-jsonschema-core" % "0.7.6",
      "com.github.andyglow" %% "scala-jsonschema-circe-json" % "0.7.6",
      "com.github.andyglow" %% "scala-jsonschema-enumeratum" % "0.7.6"
```

And specifying definitions https://github.com/andyglow/scala-jsonschema#definitionsreferences

It works but compilers show me warns:
```
method apply in trait Schema is deprecated (since will be removed in 1.0.0): 0.6.1
[warn]   implicit val myCaseClassSchema: Schema[MyCaseClass] = Json.schema[MyCaseClass]("my-case-class")
```

I'm using some old syntax? Is there something missing? How can I get rid of warns?

**Comment by andyglow** [comment]
hello, @michalbogacz 
that's the bug.. I have deprecated the api, but some macro stuff is still using it
let me fix it

**Comment by andyglow** [comment, marked as answer]
actually may i ask you to provide some example.. i double-checked the code and i fill like maybe it's on your side
the change meant by warning is `schema("my-name")` -> `schema.toDefinition("my-name")`

**Reply by michalbogacz** [reply]
If you still need example, here it is https://scastie.scala-lang.org/AOp3ADJGTOmuGS4uIQIsLw

**Reply by andyglow** [reply]
yeah. so. what needs to be done to fix warning is
```
  implicit val dataSchema: Schema[Data] = Json.schema[Data].toDefinition("data-type")
```

instead of 
```
  implicit val dataSchema: Schema[Data] = Json.schema[Data]("data-type")
```

**Reply by michalbogacz** [reply]
Thanks for workaround. I created this code looking at this doc https://github.com/andyglow/scala-jsonschema#definitionsreferences

**Reply by andyglow** [reply]
thanks for noticing that.. i've updated the readme
sorry for confusion

**Reply by andyglow** [reply]
https://github.com/andyglow/scala-jsonschema/pull/228
